src/index.js

```diff
@@ -448,12 +448,15 @@ function createConfig(options, entry, format, writeMeta) {
 						terser({
 							sourcemap: true,
 							output: { comments: false },
-							compress: {
+							compress: Object.assign({
+								passes: 10,
+								hoist_funs: true,
 								keep_infinity: true,
 								pure_getters: true,
-							},
+							}, mangleOptions && mangleOptions.compress || {}),
```

Comment:
IMHO this is confusing for uglify/terser users - because those are separate options there & not nested one like here.

Reply:
Perhaps we could use a `terser.config.json` file. It could look something like this:

```json
{
  "mangle": { ... },
  "compress": { ... }
}
```

Reply:
Well - IMHO we should support `microbundle.config.js` (or similar), personally I hate putting tool configs in package.json (but I understand some people like it), I mean tool-specific config, I'm on board with using some standard package.json fields as base for the tool config as convention. To support both we could just use https://github.com/davidtheclark/cosmiconfig

Im opposed to adding `terser.config.js` because it would suggest to users that Terser itself supports config files, which it doesnt

I plan to start a big refactoring of `microbundle` (in parts, so it would be easy to discuss, follow and review), maybe this could just sit until I do it? Otherwise we might release a feature which potentially might be changed in the near future.

Reply:
That's good to know. What are the main things you want to tackle with the refactor? As I'm new to `microbundle` I'm not up to speed on current plans.

Reply:
What I would like to tackle:
- config files
- enforce convention (need consensus on this one) with autofixer, instead of having defaults for some things (like i.e. main/module fields), just force their output filenames, but provide autofixing feature so developer can knowingly accept the changes
- validate package.json shape and certain code constructs (i.e. when detecting `typeof window`/`typeof document` enforce people to include `"browser"` field in package.json (again with autofix feature), ofc build browser bundles with those values inlined (smaller bundles 🚀)
- monorepo support
- more react-specific optimizations builtin
- when we land babel@7 support - use `@babel/runtime` helpers and offer to install that dep with autofix
- opt-in react-native bundles
- multiple entry points support (like `redux-saga` & `redux-saga/effects`) with rollup's code-splitting

Probably more 😅 plan to work on it slowly but steadily.

Reply:
Awesome, thanks for sharing! That's quite a roadmap and we should see how much we can split the tasks, so we keep being able to make timely releases 👍

I'm not sure how much time I have to contribute to those, but if I can I'd be happy to help out.

Regarding config files I'm not sure how much we want to allow users to customize besides `terser`, `babel` and `TS/Flow` options. Only the `terser` doesn't support config files on its own whereas all the others do support it already. We just have blocked it for `babel` for now until the switch away from `bublé` is complete. Perhaps we can discuss this upstream with the `terser` maintainers.

@developit Do we still need more custom `compress` settings for preact x or are the defaults you changed here good enough?

Reply:
I dont want too many things to be customizable, we have some options already though - https://github.com/developit/microbundle#all-cli-options and at least those should be supported. Also I would like to be able to override some of those options per bundle type.